webtranslate/users/github.py

```diff
@@ -0,0 +1,134 @@
+"""
+User authentication and authorization via github teams.
+
+"""
+import configparser, os, json, requests, requests_oauthlib
+from webtranslate import rights, userauth
+
+
+
+github_organization = None
+github_org_api_token = None
+github_oauth_client_id = None
+github_oauth_client_secret = None
+
+
+
+def request_teams(name):
+    """
+    Request all teams an user is member of.
+
+    @param name: Github login
+    @type  name: C{str}
+
+    @return: Teams
+    @rtype:  C{set} of C{str}
+    """
+
+    try:
+        if github_organization is not None and github_org_api_token is not None:
+            query = """query($org:String!, $user:String!) { organization(login: $org) { teams(first: 100, userLogins: [$user]) { nodes { name } } } }"""
+            variables = { "org": github_organization, "user": name }
+            r = requests.post("https://api.github.com/graphql", headers = { "Authorization": "bearer " + github_org_api_token }, json={'query': query, 'variables': variables}).json()
+            teams = set( t["name"] for t in r["data"]["organization"]["teams"]["nodes"] )
+            return teams
+        else:
+            return set()
+    except Exception as ex:
```

Comment:
This is a very broad capture of mistakes. Don't you want to make it a bit more specific?

Reply:
I guess this will become better once we add some logging framework.